tools-local/unix/search-tool.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+# Searches for the specified tool in the environment path, and the path within the repository as specified in the .toolversions file.
+
+usage()
+{
+    echo "Usage: $0 ToolName [-StrictToolVersionMatch]"
+    echo "  ToolName: Name of the tool to download."
+    echo "  (Optional) -StrictToolVersionMatch: If specified then, search will ensure that the version of the tool searched is the declared version."
+    echo "                                      Otherwise, search will attempt to find a version of the tool, which may not be the declared version."
+    echo ""
+    echo "Searches for the tool in the environment path, and the path within the repository as specified in the .toolversions file."
+    echo "If search is successful then, returns the path to the tool."
+    echo "Exit 1 if search fails to find the tool."
+}
+
+if [ $# -lt 1 ]; then
+    usage
+    exit 1
+fi
+
+if [ -z "$1" ]; then
+    echo "Argument passed as ToolName is empty. Please provide a non-empty string."
+    exit 1
+fi
+
+toolName="$1"
+shift
+strictToolVersionMatch=0
+
+while :; do
+    if [ $# -le 0 ]; then
+        break
+    fi
+
+    lowerI="$(echo $1 | awk '{print tolower($0)}')"
+    case $lowerI in
+        -stricttoolversionmatch)
+            strictToolVersionMatch=1
+            ;;
+        *)
+            usage
+            exit 1
+    esac
+    shift
+done
+
+scriptPath="$(cd "$(dirname "$0")"; pwd -P)"
+. "$scriptPath/tool-helper.sh"
+declaredVersion="$(get_tool_config_value "$toolName" "DeclaredVersion")"
+
```

Comment:
All of the logic above this point of this script seems to be duplicated to some degree across the main search/aquire/probe/invoke scripts.  Consider taking a look to see if the commonality can be factored into shared components.